uwb_platform_interface/lib/uwb_platform_interface.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+/// Contains all the calls that needs to be implemented in the platform specific plugins.
+abstract class UwbPlatform extends PlatformInterface {
+  /// Constructs a [UwbPlatform].
+  UwbPlatform() : super(token: _token);
+
+  static final Object _token = Object();
+
+  /// The instance of [UwbPlatform] to use.
+  ///
+  /// Must be set before accessing.
+  static UwbPlatform get instance => _instance;
+
+  /// Platform-specific plugins should set this with their own platform-specific.
+  /// class that extends [UwbPlatform] when they register themselves.
+  /// https://github.com/flutter/flutter/issues/43368.
+  static set instance(UwbPlatform instance) {
+    PlatformInterface.verify(instance, _token);
+    _instance = instance;
+  }
+
+  /// Should only be accessed after setter is called.
+  static late UwbPlatform _instance;
+
+  /// Checks platform compatibility.
+  ///
+  /// Returns [True] of compatible.
+  /// Returns [False] if it is not compatible.
```

Comment:
This comment should be formed as a sentence:

```suggestion
  /// Checks if the current platform supports Ultra Wide Band.
  ///
  /// Returns `true` if UWB is supported or `false` if UWB is not supported.
```